### PR TITLE
Update cloud build script to latest for v1.13.x

### PR DESCRIPTION
**Pull request by haydentherapper**
Will use the latest Go binary and Cosign version

### CODE REVIEW

**Comment by haydentherapper** [comment]
The cloud build script fails because the version of Cosign we are verifying golang-cross with can't pull down the latest metadata. Furthermore, we need to bump golang-cross to the latest (or at least a version with v2 >= 2.2.0) since we use the Cosign version installed in that container to sign the releases.

File: release/ko-sign-release-images.sh [hunk 1]
```diff
@@ -32,16 +32,8 @@ if [[ ! -f cosignImagerefs ]]; then
...
-if [[ ! -f sgetImagerefs ]]; then
-    echo "sgetImagerefs not found"
-    exit 1
-fi
-
 echo "Signing cosign images with GCP KMS Key..."
-
-cosign sign --force --key "gcpkms://projects/$PROJECT_ID/locations/$KEY_LOCATION/keyRings/$KEY_RING/cryptoKeys/$KEY_NAME/versions/$KEY_VERSION" -a GIT_HASH="$GIT_HASH" -a GIT_VERSION="$GIT_VERSION" $(cat cosignImagerefs)
-cosign sign --force --key "gcpkms://projects/$PROJECT_ID/locations/$KEY_LOCATION/keyRings/$KEY_RING/cryptoKeys/$KEY_NAME/versions/$KEY_VERSION" -a GIT_HASH="$GIT_HASH" -a GIT_VERSION="$GIT_VERSION" $(cat sgetImagerefs)
+cosign sign --yes --key "gcpkms://projects/$PROJECT_ID/locations/$KEY_LOCATION/keyRings/$KEY_RING/cryptoKeys/$KEY_NAME/versions/$KEY_VERSION" -a GIT_HASH="$GIT_HASH" -a GIT_VERSION="$GIT_VERSION" $(cat cosignImagerefs)
```
Comment (haydentherapper) [inline release/ko-sign-release-images.sh:36]: Need to update sign script too since `force` flag no longer exists

**Comment by haydentherapper** [comment]
I think I've updated everything necessary. Let me know if this can be simplified though, because this is bumping a lot of the CI infrastructure.

I _think_ we can now use the GHA to kick off the job, but if not, from the branch, we will need to run `gcloud builds submit --no-source --async --config release/cloudbuild.yaml --substitutions _GIT_TAG=v1.13.3,_TOOL_ORG=sigstore,_TOOL_REPO=cosign,_STORAGE_LOCATION=cosign-releases,_KEY_RING=release-cosign,_KEY_NAME=cosign,_GITHUB_USER=sigstore-bot --project=projectsigstore`

**Comment by haydentherapper** [comment]
I think we're good now. I didn't remove sget entirely, because that's a lot of work, but this does remove sget as a released go artifact.

**Review by cpanato — APPROVED** [review]

**Pull request merged**